evento/src/test-helper.js

```diff
@@ -0,0 +1,96 @@
+import * as enzyme from 'enzyme';
+import sinon from 'sinon';
+import Cookie from 'js-cookie';
+import fetchMock from 'fetch-mock';
+import mockFactory from './mock-factory';
+
+const cookies = {
+	set(cookies) {
+		for(let param of Object.keys(cookies)) {
+			Cookie.set(param, JSON.stringify(cookies[param]));
+		}
+	},
+
+	reset() {
+		Object.keys(Cookie.get()).forEach(function(cookie) {
+			Cookie.remove(cookie);
+		});
+	}
+};
+
+const DefaultErrorMessage = "Something went wrong";
+const mocks = {
+	// deconstructs all values (event, events, user, users etc) to this object.
+	...mockFactory.createAll(),
+
+	generate: {
+		// usage: mocks.generate.user() etc
+		...mockFactory.createAllFunctions()
+	},
+
+	api: {
+		DefaultErrorMessage: DefaultErrorMessage,
+		responses: {
+			get DefaultError() {
+				return () => { return { success: false, error: { type: "unknown", message: DefaultErrorMessage } } };
+			},
+
+			get DefaultSuccess() {
+				return () => this.create({ });
+			},
+
+			create(payload) {
+				return { success: true, payload: payload };
+			},
+
+			createError(error) {
+				return { success: false, error: { type: error.type || "unknown", message: error.message || DefaultErrorMessage } };
+			}
+		}
+	},
+};
+
+const reset = () => {
+	cookies.reset();
+	fetchMock.restore();
+};
+
+const waitForFetches = (milliseconds = 50) => {
+	return new Promise((resolve, reject) => {
+		setTimeout(() => {
+			resolve();
+		}, milliseconds);
+	});
+};
+
+const mount = async (component) => {
+	const mounted = enzyme.mount(component);
+	await waitForFetches();
+
+	mounted.wait = waitForFetches;
+	return mounted;
+};
+
+// usage: top-level "describe" in all test files uses this instead
+const test = (name, callback) => {
```

Comment:
I disagree. The wrapper contains all test reset functionality. Otherwise the exact same ~7 lines of code would have to be copy pasted into each test file, and if you forget to do that, the result can be hard-to-determine bugs in the tests.

We discussed this privately and kinda agreed to keep it.